### ISSUE TITLE
False positive: "Add or update the header of this file." when '=' characters in header.

### ISSUE THREAD

**Issue by johan162** [SonarOpenCommunity/sonar-cxx#2199]
**Describe the bug**
The CXX plugin does not recognize a file header that starts with 

```C++
/* ==========================================================================
 * 
 * ....
 * 
 */ 
```
and falsely throws a "Add or update the header of this file." code smell.

**To Reproduce**

Analyze a file that have a header like
```C++
/* =========================================================================
* File:  xxxx      
* Description: xxx
* Author:  xxxx
*
*  This program is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation, either version 3 of the License, or
*  any later version.
*
*  This program is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with this program.  If not, see <http://www.gnu.org/licenses/>
* =========================================================================
*/
```
**Expected behavior**
That the scanner recognizes this as a header and dose not raise a "Add or update the header of this file." code smell

**Desktop (please complete the following information):**
 - OS: Linux
 - SonarQube version: V8.9.1
 - cxx plugin version: V2.0.3
 - sonar-scanner version: V4.6.2

**Comment by guwirth** [comment]
Hi @johan162,

thanks for the feedback.

There was already a discussion here #2118.

* Are you using regex or plain text? Is the parameter isRegularExpression true/false?
* What encoding do your files have?

Regards,
